tests/e2e/transaction/transactionSimulation.test.ts

```diff
@@ -602,4 +602,69 @@ describe("transaction simulation", () => {
       ).rejects.toThrow();
     });
   });
+
+  describe("simulations with no account authenticator", () => {
```

Comment:
can we add more tests for all the different public key use cases? with a pk, without a pk, with array of pks (some defined some arent), etc

Reply:
I add the test cases for the multiagent transaction cases (for array of pks) : https://github.com/aptos-labs/aptos-ts-sdk/pull/467/files#diff-f97493b76e508e849a876ecb35492119c3ebf540140bda3c1a7b3313a32be2deR670-R708